executor/executor.go

```diff
@@ -455,75 +455,132 @@ func getTableName(is infoschema.InfoSchema, id int64) string {
-func (e *CheckTableExec) Next(ctx context.Context, req *chunk.Chunk) error {
-	if e.done {
-		return nil
+func (e *CheckTableExec) checkIndexHandle(ctx context.Context, num int, src *IndexLookUpExecutor) error {
+	cols := src.schema.Columns
+	retFieldTypes := make([]*types.FieldType, len(cols))
+	for i := range cols {
+		retFieldTypes[i] = cols[i].RetType
 	}
-	defer func() { e.done = true }()
-	for _, t := range e.tables {
-		dbName := t.DBInfo.Name
-		tb, err := e.is.TableByName(dbName, t.Name)
+	chk := chunk.New(retFieldTypes, e.initCap, e.maxChunkSize)
+	iter := chunk.NewIterator4Chunk(chk)
+
+	var err error
+	handles := make(map[int64]struct{}, 1024)
+	for {
+		err = src.Next(ctx, chk)
 		if err != nil {
-			return err
+			break
 		}
-		if tb.Meta().GetPartitionInfo() != nil {
-			err = e.doCheckPartitionedTable(tb.(table.PartitionedTable))
-		} else {
-			err = e.doCheckTable(tb)
+		if chk.NumRows() == 0 {
+			break
 		}
-		if err != nil {
-			logutil.Logger(ctx).Warn("check table failed", zap.String("tableName", t.Name.O), zap.Error(err))
-			if admin.ErrDataInConsistent.Equal(err) {
-				return ErrAdminCheckTable.GenWithStack("%v err:%v", t.Name, err)
+		for chunkRow := iter.Begin(); chunkRow != iter.End(); chunkRow = iter.Next() {
+			datum := chunkRow.GetDatum(0, retFieldTypes[0])
+			h := datum.GetInt64()
+			if _, ok := handles[h]; ok {
+				err = errors.Errorf("admin check table %s, index %s, at least two indices have the same handle %d more than one",
+					e.tblInfo.Name, e.indices[num].Meta().Name, h)
+				e.retCh <- errors.Trace(err)
+				return errors.Trace(err)
 			}
+			handles[h] = struct{}{}
+		}
 
-			return errors.Errorf("%v err:%v", t.Name, err)
+		select {
+		case <-e.exitCh:
+			return nil
+		default:
 		}
 	}
-	return nil
+	e.retCh <- errors.Trace(err)
+	return errors.Trace(err)
 }
 
-func (e *CheckTableExec) doCheckPartitionedTable(tbl table.PartitionedTable) error {
-	info := tbl.Meta().GetPartitionInfo()
-	for _, def := range info.Definitions {
-		pid := def.ID
-		partition := tbl.GetPartition(pid)
-		if err := e.doCheckTable(partition); err != nil {
-			return err
+// Next implements the Executor Next interface.
+func (e *CheckTableExec) Next(ctx context.Context, req *chunk.Chunk) error {
+	if e.done || len(e.srcs) == 0 {
+		return nil
+	}
+	defer func() { e.done = true }()
+
+	idxNames := make([]string, 0, len(e.indices))
+	for _, idx := range e.indices {
+		idxNames = append(idxNames, idx.Meta().Name.O)
+	}
+	greater, idxOffset, err := admin.CheckIndicesCount(e.ctx, e.dbName, e.tblInfo.Name.O, idxNames)
+	if err != nil {
+		tbl := e.srcs[idxOffset].table
+		if greater == admin.IdxCntGreater {
+			err = e.checkIndexHandle(ctx, idxOffset, e.srcs[idxOffset])
+		} else if greater == admin.TblCntGreater {
+			err = e.checkTableRecord(tbl, idxOffset)
+		}
+		if err != nil && admin.ErrDataInConsistent.Equal(err) {
+			return ErrAdminCheckTable.GenWithStack("%v err:%v", tbl.Meta().Name, err)
```

Comment:
Why you add this error check here?  We can contain the table name directly when do `admin.CheckRecordAndIndex`,  
`return ErrAdminCheckTable.GenWithStack("%v err:%v", tbl.Meta().Name, err)` 
here will lost the original stack info and then I don't know where the error returns from.

Reply:
I think it wants to unify the error message. It the original code. And the function calls `GenWithStack` it should have the original stack info.